### PR TITLE
Document data format and example for featureSpec

### PR DESCRIPTION
**Pull request by vtlim**
This PR has:
- [x] been self-reviewed.

### CODE REVIEW

File: docs/ingestion/data-formats.md [hunk 1]
```diff
@@ -74,8 +74,8 @@ parsing data is less efficient than writing a native Java parser or using an ext
 
 ## Input format
 
-> The Input Format is a new way to specify the data format of your input data which was introduced in 0.17.0.
-Unfortunately, the Input Format doesn't support all data formats or ingestion methods supported by Druid yet.
+> The `inputFormat` field is a new way to specify the data format of your input data which was introduced in 0.17.0.
```
Comment (techdocsmith) [inline docs/ingestion/data-formats.md:77]:
```suggestion
You can use the `inputFormat` field to specify the data format for your input data.
```
We're on 0.22.1. It's not new anymore. :D

File: docs/ingestion/data-formats.md [hunk 2]
```diff
@@ -74,8 +74,8 @@ parsing data is less efficient than writing a native Java parser or using an ext
...
+Unfortunately, `inputFormat` doesn't support all data formats or ingestion methods supported by Druid yet.
```
Comment (techdocsmith) [inline docs/ingestion/data-formats.md:78]:
```suggestion
> `inputFormat` doesn't support all data formats or ingestion methods supported by Druid yet.
```
Unfortunately is unnecessary.

File: docs/ingestion/data-formats.md [hunk 3]
```diff
@@ -89,7 +89,7 @@ Configure the JSON `inputFormat` to load JSON data as follows:
 |-------|------|-------------|----------|
 | type | String | Set value to `json`. | yes |
 | flattenSpec | JSON Object | Specifies flattening configuration for nested JSON data. See [`flattenSpec`](#flattenspec) for more info. | no |
-| featureSpec | JSON Object | [JSON parser features](https://github.com/FasterXML/jackson-core/wiki/JsonParser-Features) supported by Jackson library. Those features will be applied when parsing the input JSON data. | no |
+| featureSpec | JSON Object | [JSON parser features](https://github.com/FasterXML/jackson-core/wiki/JsonParser-Features) supported by Jackson, a JSON processor for Java. The features control parsing of the input JSON data. Map the feature name to a Boolean value of whether the feature is enabled. For example: `"featureSpec": {"ALLOW_SINGLE_QUOTES": true, "ALLOW_UNQUOTED_FIELD_NAMES": true}` | no |
```
Comment (techdocsmith) [inline docs/ingestion/data-formats.md:92]:
```suggestion
| featureSpec | JSON Object | [JSON parser features](https://github.com/FasterXML/jackson-core/wiki/JsonParser-Features) supported by Jackson, a JSON processor for Java. The features control parsing of the input JSON data. To enable a feature, map the feature name to a Boolean value of "true". For example: `"featureSpec": {"ALLOW_SINGLE_QUOTES": true, "ALLOW_UNQUOTED_FIELD_NAMES": true}` | no |
```
If a feature is not listed as "true", does it default to "false"? Maybe not for this PR, but do we need a list of features?

**Review by techdocsmith — CHANGES_REQUESTED** [review]
Minor changes.

File: docs/ingestion/data-formats.md [hunk 4]
```diff
@@ -584,7 +584,7 @@ For example:
 
 ### FlattenSpec
 
-The `flattenSpec` bridges the gap between potentially nested input data (such as JSON, Avro, etc) and Druid's flat data model. It is an object within the `inputFormat` object.
+The `flattenSpec` object bridges the gap between potentially nested input data (such as JSON, Avro, etc) and Druid's flat data model. It is an object within the `inputFormat` object.
```
Comment (techdocsmith) [inline docs/ingestion/data-formats.md:587]: nit: avoid parens if possible

File: docs/ingestion/data-formats.md (hunk 3 above)
Comment (vtlim) [reply docs/ingestion/data-formats.md:92]: The description links to the Jackson parser doc, where the features are listed.

**Pull request merged**